src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -525,15 +525,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
 
   <Target Name="RenamePublishedHost"
+          DependsOnTargets="_ComputeNETCoreBuildOutputFiles"
           AfterTargets="ComputeFilesToPublish"
           BeforeTargets="CopyFilesToPublishDirectory"
           Condition="'$(RenamePublishedHost)' == 'true'">
 
     <ItemGroup>
 
-      <ResolvedFileToPublish Condition="'%(ResolvedFileToPublish.RelativePath)' == '$(_DotNetHostExecutableName)'">
-        <RelativePath>$(AssemblyName)%(ResolvedFileToPublish.Extension)</RelativePath>
-      </ResolvedFileToPublish>
+      <ResolvedFileToRemove  Include ="%(ResolvedFileToPublish.Identity)" Condition="'%(ResolvedFileToPublish.RelativePath)' == '$(_DotNetHostExecutableName)'"/>
+      <ResolvedFileToPublish Remove ="%(ResolvedFileToRemove.Identity)"/>
+
+      <ResolvedFileToAdd Include="%(NativeAppHostNETCore.Identity)">
```

Comment:
No it doesn't.  Just like:

```
<Content Include="File1">
</Content>

<Content Include="File2">
  <CopyToOutputDirectory>Always</CopyToOutputDirectory>
</Content>
```

In that scenario, how many files get copied to the output directory?  Only `File2`.  Adding a new Item and setting metadata on the item doesn't set the metadata on all the existing files in that item.